src/passkey_child/passkey_child_common.c

```diff
@@ -319,3 +322,71 @@ register_key(struct passkey_data *data)
 
     return ret;
 }
+
+errno_t
+public_key_to_base64(TALLOC_CTX *mem_ctx, const struct passkey_data *data,
+                     const unsigned char *public_key, size_t pk_len,
+                     char **_pem_key)
+{
+    EVP_PKEY *evp_pkey = NULL;
+    unsigned char *pub = NULL;
+    char *pem_key = NULL;
+    unsigned long err;
+    errno_t ret;
+
+    switch (data->type) {
+    case COSE_ES256:
+        ret = es256_pubkey_to_evp_pkey(mem_ctx, public_key, pk_len, &evp_pkey);
+        break;
+    case COSE_RS256:
+        ret = rs256_pubkey_to_evp_pkey(mem_ctx, public_key, pk_len, &evp_pkey);
+        break;
+    case COSE_EDDSA:
+        ret = eddsa_pubkey_to_evp_pkey(mem_ctx, public_key, pk_len, &evp_pkey);
+        break;
+    default:
+        DEBUG(SSSDBG_OP_FAILURE, "Invalid key type.\n");
+        ret = EINVAL;
+        break;
+    }
+
+    if (ret != EOK) {
+        goto done;
+    }
+
+    ret = i2d_PUBKEY(evp_pkey, &pub);
+    if (ret < 1) {
+        err = ERR_get_error();
+        DEBUG(SSSDBG_OP_FAILURE, "i2d_PUBKEY failed [%lu][%s].\n",
+              err, ERR_error_string(err, NULL));
+        ret = EIO;
+        goto done;
+    }
+
+    pem_key = sss_base64_encode(mem_ctx, pub, ret);
+    if (pem_key == NULL) {
+        DEBUG(SSSDBG_OP_FAILURE, "sss_base64_encode failed.\n");
+        ret = ENOMEM;
+        goto done;
+    }
+
+    if (_pem_key != NULL) {
+        *_pem_key = talloc_strndup(mem_ctx, pem_key, strlen(pem_key));
```

Comment:
Why do you need a copy?

Reply:
No need for it. Fixed.

Reply:
`if (_pem_key != NULL)` check made some sense.

Reply:
Added the check at the beginning of the function.